Web.Client/Pages/Admin/GradeRegistrationCriteriaEdit.razor.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Havit;
+using MensaGymnazium.IntranetGen3.Contracts;
+using MensaGymnazium.IntranetGen3.Web.Client.Services.DataStores;
+
+namespace MensaGymnazium.IntranetGen3.Web.Client.Pages.Admin;
+
+public partial class GradeRegistrationCriteriaEdit
+{
+	[Parameter] public GradeRegistrationCriteriaDto Value { get; set; }
+	[Parameter] public EventCallback<GradeRegistrationCriteriaDto> ValueChanged { get; set; }
+	[Parameter] public EventCallback OnClosed { get; set; }
+
+	[Inject] protected IGradeFacade GradeFacade { get; set; }
+	[Inject] protected IGradesDataStore GradesDataStore { get; set; }
+
+	private GradeRegistrationCriteriaDto model = new();
+	private EditContext editContext;
+	private HxOffcanvas hxOffcanvas;
+
+	protected override Task OnInitializedAsync()
+	{
+		return GradesDataStore.EnsureDataAsync();
+	}
+
+	protected override void OnParametersSet()
+	{
+		base.OnParametersSet();
```

Comment:
v Blazoru se kupodivu volání `base.Xy()` moc nedrží, pokud se dědí přímo z `ComponentBase`
dokonce oficiální dokumentace to zmiňuje, že to není potřeba
(ty metody mají prázdnou implementaci)